### PR TITLE
Servicenow advanced rules

### LINKED ISSUE #948: ServiceNow Advanced Sync Rules
Similarly to the advanced filtering capabilities added to Jira and Confluence e.g. https://github.com/elastic/connectors-python/issues/874, support customized filtering via advanced sync rules for ServiceNow.

### Acceptance criteria
- [ ] #1112
- [ ] add documentation with several examples illustrating the syntax, how a basic use case works, limitations
- [ ] document impact of advanced sync rules on connector configuration and on document indexing
- [ ] add tooltip on the configurable fields that are ignored when advanced sync rules are used, informing the user about the behavior

### PR DESCRIPTION
**Pull request by parth-elastic**
## Closes #948 #1112 

1. This PR introduces advanced filtering rules for ServiceNow by queries to filter out document based on that
2. Added documentation with several examples illustrating the syntax, how a basic use case works, limitations
3. Added documentation regarding impact of advanced sync rules on connector configuration and on document indexing
4. Included tooltip on the configurable fields that are ignored when advanced sync rules are used

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] if there is no GH issue, please create it. Each PR should have a link to an issue
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)
- [x] Considered corresponding documentation changes
- [x] Contributed any configuration settings changes to the configuration reference

### CODE REVIEW

File: connectors/sources/servicenow.py [hunk 1]
```diff
@@ -369,6 +400,65 @@ async def close_session(self):
         del self._get_session
 
 
+class ServiceNowAdvancedRulesValidator(AdvancedRulesValidator):
+    RULES_OBJECT_SCHEMA_DEFINITION = {
+        "type": "object",
+        "properties": {
+            "service": {"type": "string", "minLength": 1},
```
Comment (timgrein) [inline connectors/sources/servicenow.py:407]: Can `service` be an arbitrary string or must it be a specific element of a specific set of strings?

**Review by timgrein — COMMENTED** [review]
Looks good! Asked one question around validation

File: connectors/sources/servicenow.py (hunk 1 above)
Comment (parth-elastic) [reply connectors/sources/servicenow.py:407]: Only a `set of alphabets` is acceptable for valid service.

**Review by timgrein — APPROVED** [review]
LGTM

**Pull request merged**